### DISCUSSION TITLE
Problems creating a case with only renewable generators

### DISCUSSION THREAD

**Discussion by amparocorvalan** [CURENT/andes#564] [Modeling] [answered]
Dear @cuihantao and @jinningwang,

I am working on a case study consisting of a **wind plant**, a **battery** and a series of **loads** (modeling an off-grid system). 

In a first instance I simulated the same system but on-grid, including a synchronous generator. By eliminating the synchronous generator the system fails to calculate the power flow. I have tried this in different ways:

1. replacing all ieee14 synchronous generators by renewable generators.
2. build a small and simple case of only renewable generators and loads. 
3. create the case I need to study, consisting of renewable generators, batteries and loads. 

In all these attempts the problem was always the same, the **Jacobian matrix is singular** and the power flow calculation does not continue.

![image](https://github.com/user-attachments/assets/5150ac7c-5bd3-4a37-9724-b96a3032b49d)

The only way I have been able to get these systems to run is to leave the synchronous generator connected but without load (as what could be a synchronous capacitor).

Can you help me to solve this problem? Is it possible to create in ANDES cases that do not have synchronous generators? 

I would be very grateful for your valuable support. Solving this would be a great help in order for me to complete my master thesis.

**Comment by cuihantao** [comment]
Hi, @amparocorvalan 

The singular Jacobian matrix during power flow may be caused by something simple. The way to troubleshoot is to delete all the models that are not used in power flow, so as to narrow down the issue. Dynamic generators, e.g., GENROU, are not used in power flow; they are initialized after solving it. 

Are there any new models that you developed being used in power flow?

**Reply by amparocorvalan** [reply]
Thank you very much for your reply @cuihantao!

I corrected a bug I had and now the power flow does converge. However, when doing a TDS, I have the same problem again with the singular matrix. 

To test making a full renewable system, I created a base case consisting of WTs, a GS and loads. For this case the power flow and TDS converge. 

However, by modifying this case so that the GS is replaced by WTs, no solution is found and the matrix is singular:
![image](https://github.com/user-attachments/assets/d68669fd-d244-4d1d-9295-f57d1570e08f)

Have you been able to create a full renewable case? 

The base cases I am using to test if it works are the following:
case with one SG: 
[base_case_ongrid_REN.xlsx](https://github.com/user-attachments/files/16715028/base_case_ongrid_REN.xlsx)
case without SG: 
[base_case_offgrid_REN.xlsx](https://github.com/user-attachments/files/16715036/base_case_offgrid_REN.xlsx)

**Comment by cuihantao** [comment]
I did get a fully renewable system working. The only trick is to retain a Slack so that the system has a reference for angle. This might solve your issue because the output does not report any all-zero rows/zeros at the main diagonal.

**Reply by amparocorvalan** [reply]
What do you mean by retaining a slack? Do you mean using a grid forming (for example) to give the reference (as there are no synchronous machines)? How did you define your slack bus?

Thank you again for your reply!

**Reply by cuihantao** [reply, marked as answer]
You are welcome. When creating the case for power flow, you have one Slack device for each island. You can keep the Slack there without using any dynamic generator to substitute it. See if that gets the simulation to go thorugh.

**Reply by amparocorvalan** [reply]
Thank you very much for your help and time to answer my questions! It worked leaving the slack bus without dynamic generator.
